sdk/peering/Azure.ResourceManager.Peering/tests/Scenario/PeeringServiceTests.cs

```diff
@@ -0,0 +1,119 @@
+using Azure.ResourceManager.Peering.Models;
+using Azure.ResourceManager.Resources;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Peering.Tests
+{
+    internal class PeeringServiceTests : PeeringManagementTestBase
+    {
+        private ResourceGroupResource _resourceGroup;
+        private PeeringServiceCollection _peeringServiceCollection => _resourceGroup.GetPeeringServices();
+
+        public PeeringServiceTests(bool isAsync) : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public async Task TestSetUp()
+        {
+            _resourceGroup = await CreateResourceGroup();
+        }
+
+        [RecordedTest]
+        public async Task CreateOrUpdate()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            var peeringService = await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            ValidatePeeringService(peeringService);
+            Assert.AreEqual(peeringServiceName, peeringService.Data.Name);
+        }
+
+        [RecordedTest]
+        public async Task Exist()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            bool flag = await _peeringServiceCollection.ExistsAsync(peeringServiceName);
+            Assert.IsTrue(flag);
+        }
+
+        [RecordedTest]
+        public async Task Get()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            var peeringService = await _peeringServiceCollection.GetAsync(peeringServiceName);
+            ValidatePeeringService(peeringService);
+            Assert.AreEqual(peeringServiceName, peeringService.Value.Data.Name);
+        }
+
+        [RecordedTest]
+        public async Task GetAll()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            var list = await _peeringServiceCollection.GetAllAsync().ToEnumerableAsync();
+            Assert.IsNotEmpty(list);
+            ValidatePeeringService(list.First(item => item.Data.Name == peeringServiceName));
+        }
+
+        [RecordedTest]
+        public async Task Delete()
+        {
+            string peeringServiceName = Recording.GenerateAssetName("peeringService");
+            var peeringService = await CreateAtmanPeeringService(_resourceGroup, peeringServiceName);
+            bool flag = await _peeringServiceCollection.ExistsAsync(peeringServiceName);
+            Assert.IsTrue(flag);
+
+            await peeringService.DeleteAsync(WaitUntil.Completed);
+            flag = await _peeringServiceCollection.ExistsAsync(peeringServiceName);
+            Assert.IsFalse(flag);
+        }
+
+        [RecordedTest]
```

Comment:
Can we follow this [pattern](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/storage/Azure.ResourceManager.Storage/tests/Tests/StorageAccountTests.cs#L1258-L1260) to test the tag operation?